prompt_toolkit/widgets/base.py

```diff
@@ -508,6 +509,22 @@ def _(event):
         def _(event):
             self.current_value = self.values[self._selected_index][0]
 
+        @kb.add(Keys.Any)
+        def _(event):
+            if six.binary_type(self.values[self._selected_index][1]).startswith(event.data):
+                # In the case we are already on a value starting by the key: seek for the next one
+                try:
+                    self._selected_index = self.values.index(next(x for x in self.values[self._selected_index + 1:]
```

Comment:
For readability, I'd prefer a normal for-loop with a break, instead of using ``next``. Something like:
```python
for value in self.values[self._selected_index + 1:] + self.values:
    if value.startswith(event.data):
        self._selected_index = self.values.index(value)
        break
```
Adding ``self.values`` here helps searching again from the beginning.

Reply:
Good idea for the `+ self.values` ! Updated